lib/mayaUsd/ufe/MayaUsdHierarchyHandler.cpp

```diff
@@ -50,5 +53,90 @@ Ufe::Hierarchy::Ptr MayaUsdHierarchyHandler::hierarchy(const Ufe::SceneItem::Ptr
     return MayaUsdHierarchy::create(usdItem);
 }
 
+#ifdef MAYAUSD_NEED_OUTLINER_FILTER_FIX
+
+static std::map<std::string, bool> fsCachedFilterValues = {
+    { "InactivePrims", true },
+    { "ClassPrims", false },
+};
+
+static std::atomic<bool> hasPendingFilterDefaultsUpdate = false;
+
+static void updateFilterDefaults(void*)
+{
+    // This script finds a valid outiner panel that has valid UFE filters
+    // settings.
```

Comment:
What happens if there are multiple Outliners open? How do you return the correct result for the Outliner you are being called from?

Reply:
The children filter value kept by the outliner are in a static variable, shared by all outliners. There is only one set of value, so it does not matter which outliner we use. I only need to go through the list of them all because Maya creates a second hidden outliner on start-up, which is invalid until first shown but returned as part of the list of outliner panels.

Reply:
After further testing, turns out I had tested with local Maya 2025 build with the fix for multiple-filter in it, so this script does not actually fully solve the problem. The problem is when loading a Maya scene, the filters are not restored correctly in Maya 2025. It might turn out that we can't support class prims filter in Maya 2025...

Reply:
I fixed that problem by not calling the script on-idle but calling it right away.